gocourse04/smart_zoo_test.go

```diff
@@ -0,0 +1,218 @@
+	area := NewArea("area_1", "Happy birds", "Feathered")
+	sector := NewSector("sector_1", "Parrots", NewTechnicalRoom("tr_1"))
+	area.AddSector(sector)
+
+	_, exists := area.GetSector("sector_1")
+	assert.True(t, exists, "Sector shouldn't exist")
+
+	area.DeleteSector("sector_1")
+	_, exists = area.GetSector("sector_1")
+	assert.False(t, exists, "Sector shouldn't exist")
+}
+
+func TestAddAnimal(t *testing.T) {
+	sector := NewSector("sector_1", "Parrots", NewTechnicalRoom("tr_1"))
+	animal := NewAnimal("animal_1", "Frodo")
+	sector.AddAnimal(animal)
+
+	s, exists := sector.GetAnimal("animal_1")
+	assert.True(t, exists, "Animal should exist")
+	assert.Equal(t, animal, s, "Animals should be equal")
+}
+
+func TestDeleteAnimal(t *testing.T) {
+	sector := NewSector("sector_1", "Parrots", NewTechnicalRoom("tr_1"))
+	animal := NewAnimal("animal_1", "Frodo")
+	sector.AddAnimal(animal)
+
+	_, exists := sector.GetAnimal("animal_1")
+	assert.True(t, exists, "Animal should exist")
+
+	sector.DeleteAnimal("animal_1")
+	_, exists = sector.GetAnimal("animal_1")
+	assert.False(t, exists, "Animal shouldn't exist")
+}
+
+func TestAddArea(t *testing.T) {
+	zoo := NewZoo("Happy zoo")
+	area := NewArea("area_1", "Happy birds", "Feathered")
+	zoo.AddArea(area)
+
+	z, exists := zoo.GetArea("area_1")
+	assert.True(t, exists, "Area should exist")
+	assert.Equal(t, area, z, "Areas should be equal")
+}
+
+func TestDeleteArea(t *testing.T) {
+	zoo := NewZoo("Happy zoo")
+	area := NewArea("area_1", "Happy birds", "Feathered")
+	zoo.AddArea(area)
+
+	_, exists := zoo.GetArea("area_1")
+	assert.True(t, exists, "Area should exist")
+
+	zoo.DeleteArea("area_1")
+	_, exists = zoo.GetArea("area_1")
+	assert.False(t, exists, "Area shouldn't exist")
+}
+
+func TestLookupAnimalByID(t *testing.T) {
+	area := NewArea("area_1", "Happy birds", "Feathered")
+	sector := NewSector("sector_1", "Parrots", nil)
+	sector2 := NewSector("sector_2", "Ducks", nil)
+	animal := NewAnimal("animal_1", "Frodo")
+	animal2 := NewAnimal("animal_2", "Bilbo")
+	sector.AddAnimal(animal)
+	sector.AddAnimal(animal2)
+	area.AddSector(sector)
+	area.AddSector(sector2)
+	zoo := NewZoo("Happy zoo")
+	zoo.AddArea(area)
+
+	sectorFound, ok := zoo.LookupAnimalByID("animal_2")
+
+	assert.True(t, ok, "Animal should be found")
+	assert.Equal(t, sectorFound, sector, "Sectors should be equal")
+}
+
+func TestFailLookupAnimalByID(t *testing.T) {
+	area := NewArea("area_1", "Happy birds", "Feathered")
+	sector := NewSector("sector_1", "Parrots", nil)
+	sector2 := NewSector("sector_2", "Ducks", nil)
+	animal := NewAnimal("animal_1", "Frodo")
+	animal2 := NewAnimal("animal_2", "Bilbo")
+	sector.AddAnimal(animal)
+	sector.AddAnimal(animal2)
+	area.AddSector(sector)
+	area.AddSector(sector2)
+	zoo := NewZoo("Happy zoo")
+	zoo.AddArea(area)
+
+	sectorFound, ok := zoo.LookupAnimalByID("animal_3")
+
+	assert.False(t, ok, "Animal should be found")
+	assert.Equal(t, sectorFound, Sector{}, "Sectors should be equal")
```

Comment:
див. попередній комент:
```suggestion
	assert.Equal(t, Sector{}, sectorFound, "Sectors should be equal")
```